packages/component-base/src/data-provider-controller/helpers.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Retreives information for the given flattened index, including:
+ * - the corresponding cache
+ * - the associated item (if loaded)
+ * - the corresponding index in the cache's items array.
+ * - the page containing the index.
+ * - the cache level
+ *
+ * @param {import('./cache.js').Cache} cache
+ * @param {number} flatIndex
+ * @return {{ cache: Cache, item: object | undefined, index: number, page: number, level: number }}
+ */
+export function getFlatIndexInfo(cache, flatIndex, level = 0) {
```

Comment:
**note:** Replaces `ItemCache.getCacheAndIndex` and `ItemCache.getItemForIndex`

https://github.com/vaadin/web-components/blob/f1cea1bcb1d4bd2fe1f16390c452e2b2131823b2/packages/grid/src/vaadin-grid-data-provider-mixin.js#L54-L57

https://github.com/vaadin/web-components/blob/f1cea1bcb1d4bd2fe1f16390c452e2b2131823b2/packages/grid/src/vaadin-grid-data-provider-mixin.js#L86-L98

Reply:
**question:** Should these functions be separate helpers or should they be methods of the `Cache` class?